addons/sale/models/sale.py

```diff
@@ -402,6 +403,28 @@ def _onchange_commitment_date(self):
                 }
             }
 
+    @api.onchange('pricelist_id', 'order_line')
+    def _onchange_pricelist_id(self):
+        if self.order_line and self._origin.pricelist_id and self._origin.pricelist_id != self.pricelist_id:
+            self.is_change_pricelist = True
+        else:
+            self.is_change_pricelist = False
+
+    def update_pricelist(self):
+        self.ensure_one()
+        for line in self.order_line:
```

Comment:
`for line in self.order_line.filtered(lambda line: not line.display_type)`
We do not want to recompute prices on section/note lines ;).

Reply:
@Feyensv Done :)